lib/gyp/generator/ninja/index.js

```diff
@@ -573,17 +583,22 @@ NinjaMain.prototype.vars = function vars() {
     cc = 'clang';
     cxx = 'clang++';
   } else if (process.platform === 'win32') {
-    ar = 'lib.exe';
-    ld = 'link.exe';
+    ar = gyp.bindings.which('lib.exe');
```

Comment:
That's the point of environment files. If it doesn't work - would you mind posting results of `which link.exe` and the environment file itself?